tests/integration/test_inject_python_args.py

```diff
@@ -0,0 +1,262 @@
+    boot_mode_args,  # type: List[str]
+    exe,  # type: str
+):
+    # type: (...) -> None
+
+    pex = os.path.join(str(tmpdir), "pex")
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+
+    run_pex_command(
+        args=execution_mode_args
+        + boot_mode_args
+        + [
+            "--pex-root",
+            pex_root,
+            "--runtime-pex-root",
+            pex_root,
+            "--exe",
+            exe,
+            "--inject-python-args=-W ignore",
+            "-o",
+            pex,
+        ]
+    ).assert_success()
+
+    assert_exe_output(pex, warning_expected=False)
+    assert_exe_output(pex, warning_expected=False)
+
+    # N.B.: The original argv cannot be detected by Pex running under PyPy<3.10; so we expect
+    # warnings to be turned off (the default sealed in by `--inject-python-args`). For all other
+    # Pythons we support, these explicit command line Python args should be detected and trump the
+    # injected args by dint of occurring later in the command line. In other words, the command line
+    # should be as follows and Python is known to pick the last occurrence of the -W option:
+    #
+    #   python -W ignore -W always ...
+    #
+    warning_expected = not IS_PYPY or sys.version_info[:2] >= (3, 10)
+    assert_exe_output(
+        pex, prefix_args=[sys.executable, "-W", "always"], warning_expected=warning_expected
+    )
+    assert_exe_output(
+        pex, prefix_args=[sys.executable, "-W", "always"], warning_expected=warning_expected
+    )
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] < (3, 10 if IS_PYPY else 9),
+    reason=(
+        "The effect of `-u` on the `sys.stdout.buffer` type used in this test is only "
+        "differentiable from a lack of `-u` for these Pythons."
+    ),
+)
+@parametrize_execution_mode_args
+@parametrize_boot_mode_args
+def test_issue_2422(
+    tmpdir,  # type: Any
+    execution_mode_args,  # type: List[str]
+    boot_mode_args,  # type: List[str]
+):
+    # type: (...) -> None
+
+    pex = os.path.join(str(tmpdir), "pex")
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+
+    exe = os.path.join(str(tmpdir), "exe.py")
+    with open(exe, "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                import sys
+
+
+                if __name__ == "__main__":
+                    print(type(sys.stdout.buffer).__name__)
+                """
+            )
+        )
+
+    args = ["--pex-root", pex_root, "--runtime-pex-root", pex_root, "--exe", exe, "-o", pex]
+    args.extend(execution_mode_args)
+    args.extend(boot_mode_args)
+
+    run_pex_command(args=args).assert_success()
+    shutil.rmtree(pex_root)
+    assert b"BufferedWriter\n" == subprocess.check_output(args=[sys.executable, exe])
+    assert b"BufferedWriter\n" == subprocess.check_output(
+        args=[pex]
+    ), "Expected cold run to use buffered io."
+    # assert b"BufferedWriter\n" == subprocess.check_output(
```

Comment:
This seems like some stray code that can be deleted.

Reply:
Yup! Actually, un-commented. I want to be running all these 2x in a row to hit the cold cache, then the warm cache cases, which re-direct through different `execv` code paths.